### PR TITLE
Fix OSX hyperkube packaging with updated "mktemp -d" usage

### PR DESCRIPTION
**Pull request by ikatson**
**What this PR does / why we need it**:

Before this patch, the ```make release``` command does not finish successfully. The reason is the ```kube::release::package_hyperkube``` can't succeed, because the usage of ```mktemp -d``` needs to be updated for OSX version of ```mktemp```

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)*:
didn't find any existing issues

**Special notes for your reviewer**:

**Release note**:

```release-note
```
The PR title sounds good enough for the release note

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36692)

### CODE REVIEW

**Comment by ikatson** [comment]
I signed it!

**Comment by jayunit100** [comment]
whats the `XXXXXXX` supposed to be?

**Comment by ikatson** [comment]
Thanks @jayunit100 , turns out it's not needed. I started testing this without the "-t" flag, where you need to exec smth like ```mktemp -d /tmp/hyperkube.XXXXXX```

File: cluster/images/hyperkube/Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 REGISTRY?=gcr.io/google_containers
 ARCH?=amd64
-TEMP_DIR:=$(shell mktemp -d)
+TEMP_DIR:=$(shell mktemp -d 2>/dev/null || mktemp -d -t hyperkube)
```
Comment (mikedanese) [inline cluster/images/hyperkube/Makefile]: just make this
```
TEMP_DIR:=$(shell mktemp -d -t hyperkube 2>/dev/null)
```

Comment (ikatson) [inline cluster/images/hyperkube/Makefile]: Does not work on linux for me:

```
[root@6f4669653875 /]# mktemp -d -t hyperkube
mktemp: too few X's in template 'hyperkube'
```

This one ```mktemp -d -t hyperkubeXXXXXX``` works on both though, though the XXX part is present on OSX in the resulting file, e.g.:

```
mktemp -d -t hyperkubeXXXXXX
/var/folders/c_/jc02k3ls0sggs3hc8h4qltyxj_5h0_/T/hyperkubeXXXXXX.mFkrScke
```

I'll just change to this one then, cause it's probably not a big deal.

**Comment by mikedanese** [comment]
please squash commits then lgtm

**Comment by ikatson** [comment]
@mikedanese done

**Comment by mikedanese** [comment]
@k8s-bot gci gce e2e test this

**Pull request merged**